serializer/serializer.c

```diff
@@ -673,6 +673,102 @@ status_t receive_mam_message_res_serialize(char* const message, char** obj) {
   return ret;
 }
 
+status_t send_mam_req_deserialize(const char* const obj, ta_send_mam_req_t* req) {
+  if (obj == NULL) {
+    ta_log_error("%s\n", "SC_SERIALIZER_NULL");
+    return SC_SERIALIZER_NULL;
+  }
+  cJSON* json_obj = cJSON_Parse(obj);
+  cJSON* json_result = NULL;
+  status_t ret = SC_OK;
+
+  if (json_obj == NULL) {
+    ret = SC_SERIALIZER_JSON_PARSE;
+    ta_log_error("%s\n", "SC_SERIALIZER_JSON_PARSE");
+    goto done;
+  }
+
+  json_result = cJSON_GetObjectItemCaseSensitive(json_obj, "seed");
+  if (json_result != NULL) {
+    size_t seed_size = strlen(json_result->valuestring);
+
+    if (seed_size != NUM_TRYTES_HASH) {
+      ret = SC_SERIALIZER_INVALID_REQ;
+      ta_log_error("%s\n", "SC_SERIALIZER_INVALID_REQ");
+      goto done;
+    }
+    req->seed = (tryte_t*)malloc(sizeof(tryte_t) * NUM_TRYTES_ADDRESS + 1);
+    snprintf((char*)req->seed, seed_size + 1, "%s", json_result->valuestring);
+  }
+
+  json_result = cJSON_GetObjectItemCaseSensitive(json_obj, "channel_ord");
+  if ((json_result != NULL) && cJSON_IsNumber(json_result)) {
+    req->channel_ord = json_result->valueint;
+  }
+
+  json_result = cJSON_GetObjectItemCaseSensitive(json_obj, "message");
+  if (json_result != NULL) {
+    size_t payload_size = strlen(json_result->valuestring);
```

Comment:
Does message have size limit?

Reply:
No it doesn't
But it must be in reasonable range.

Reply:
Maybe we can set up a limit ourselves? :thinking:

Reply:
Maybe we could do that. However, I think we may dicuss this later, since the scenario of our MAM application is not that clear at this moment.
And if we implement this limit, we would meet some barrier when we are going to develop recveiving MAM message API

Reply:
@HowJMay, create a new issue for the constraints.

Reply:
I did. It's #325